components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java

```
@@ -253,6 +266,33 @@ public boolean validateScopes(String username, OpenAPI openAPI, MessageContext s
         throw new APISecurityException(APISecurityConstants.INVALID_SCOPE, "Scope validation failed");
     }
 
+    private String[] getUserRoles(String username) throws APISecurityException {
+        String[] userRoles;
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+        if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            try {
+                userRoles = remoteUserStoreManagerServiceStub
+                        .getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(username));
+            } catch (RemoteException | RemoteUserStoreManagerServiceUserStoreExceptionException e) {
+                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, e.getMessage());
+            }
+        } else {
+            try {
+                int tenantId = ServiceReferenceHolder.getInstance()
+                        .getRealmService().getTenantManager().getTenantId(tenantDomain);
+
+                UserStoreManager manager = ServiceReferenceHolder
+                        .getInstance().getRealmService()
+                        .getTenantUserRealm(tenantId).getUserStoreManager();
+
+                userRoles = manager.getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(username));
+            } catch (UserStoreException e) {
+                throw new APISecurityException(APISecurityConstants.API_AUTH_GENERAL_ERROR, e.getMessage());
```

Swallowing exception

---

done